### PR TITLE
Full recipe design

### PR DESCRIPTION
**Pull request by Fukuyau**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java [hunk 1]
```diff
@@ -82,31 +97,120 @@ public void createInstanceOfRateRecipeFragmentDoesNotThrowError(){
...
+        assertNotNull(images);
+        assertEquals(5, ratingBar.getRate());
+        assertEquals((int)images.get(0).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(1).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(2).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(3).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(4).getTag(), ratingBar.getFullImageResource());
+    }
+
+    private void rateCurrentRecipeNStars(int nbStars){
```
Comment (wilderino) [inline app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java:183]: You should test that something happens after the click

**Review by wilderino — APPROVED** [review]
Looks very nice, maybe you could consider my comment
You should also move the modify button

File: app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java [hunk 2]
```diff
@@ -82,31 +97,120 @@ public void createInstanceOfRateRecipeFragmentDoesNotThrowError(){
...
+                break;
+            case 5:
+                onView(withId(R.id.star4)).perform(click());
+                break;
+            default:
+                break;
+        }
     }
 
     private synchronized void sendRateAndCheckToast(String expectedText) throws InterruptedException {
```
Comment (ChapatteMateo) [inline app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java:208]: Ça ne devrait plus être flaky maintenant avec Cirrus
Et il ne faut pas que ça reste dans la version finale de l'apps, pcq ils vont checker que les tests soit meaningful

**Review by ChapatteMateo — CHANGES_REQUESTED** [review]

**Review by ChapatteMateo — APPROVED** [review]

File: app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java (hunk 1 above)
Comment (Fukuyau) [reply app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java:183]: But I actually do in the image resource tests when I test that the correct image has been displayed meaning the click works ?

**Pull request merged**